jerry-core/jerry.cpp

```diff
@@ -899,6 +899,64 @@ bool jerry_api_get_object_field_value (jerry_api_object_t *object_p,
 }
 
 /**
+ * Applies the given function to the every fields in the objects
+ *
+ */
+void
+jerry_api_foreach_object_field(jerry_api_object_t *object_p, /**< object */
+                               jerry_object_field_foreach_t foreach_p, /**< foreach function */
+                               void *user_data) /**< user data for foreach function */
```

Comment:
if someone need keep there context in foreach callback, the argument passed to callback function is necessary.
for example
```cpp
Foobar foo;
auto foreach = [] (const jerry_api_string_t *name, const jerry_api_value_t *value, void *user_data)
{
    Foobar *foo = reinterpret_cast<Foobar*>(user_data);
    foo->somefunction(name, value);
}
jerry_api_foreach_object_field(obj, foreach, &foo);
                                                            
```

if callback function pointer can be replaced to std::function, user data does not need. (std::function can carry other data) but I think this project avoid c++ STL by design.